log/DESIGN.md

```diff
@@ -0,0 +1,198 @@
+
+The Go module consits of the following packages:
+
+- `go.opentelemetry.io/otel/log`
+- `go.opentelemetry.io/otel/log/embedded`
+- `go.opentelemetry.io/otel/log/noop`
+
+### LoggerProvider
+
+The [`LoggerProvider` abstraction](https://opentelemetry.io/docs/specs/otel/logs/bridge-api/#loggerprovider)
+is defined as an interface.
+
+```go
+type LoggerProvider interface{
+	embedded.LoggerProvider
+    Logger(name string, options ...LoggerOption) Logger
+}
+```
+
+### Logger
+
+The [`Logger` abstraction](https://opentelemetry.io/docs/specs/otel/logs/bridge-api/#logger)
+is defined as an interface.
+
+```go
+type Logger interface{
+	embedded.Logger
+    Emit(ctx context.Context, options ...RecordOption)
+}
+```
+
+The `Logger` has `Emit(context.Context, options ...RecordOption` method.
+
+The options are used to set log record parameters e.g. `WithBody`, `WithTimestamp`.
+There would NOT be a `WithLogRecord` option.
+
+### Record
+
+The [`LogRecord` abstraction](https://opentelemetry.io/docs/specs/otel/logs/bridge-api/#logger)
+is defined as a struct.
+
+```go
+type Record struct {
+	Timestamp         time.Time
+	ObservedTimestamp time.Time
+	Severity          Severity
+	SeverityText      string
+	Body              string
+
+	// Allocation optimization: an inline array sized to hold
+	// the majority of log calls (based on examination of open-source
+	// code). It holds the start of the list of Attrs.
+	front [nAttrsInline]attribute.KeyValue
+
+	// The number of Attrs in front.
+	nFront int
+
+	// The list of Attrs except for those in front.
+	// Invariants:
+	//   - len(back) > 0 iff nFront == len(front)
+	//   - Unused array elements are zero. Used to detect mistakes.
+	back []attribute.KeyValue
+}
+
+const nAttrsInline = 5
+
+type Severity int
+
+const (
+	SeverityUndefined Severity = iota
+	SeverityTrace
+	SeverityTrace2
+	SeverityTrace3
+	SeverityTrace4
+	SeverityDebug
+	SeverityDebug2
+	SeverityDebug3
+	SeverityDebug4
+	SeverityInfo
+	SeverityInfo2
+	SeverityInfo3
+	SeverityInfo4
+	SeverityWarn
+	SeverityWarn2
+	SeverityWarn3
+	SeverityWarn4
+	SeverityError
+	SeverityError2
+	SeverityError3
+	SeverityError4
+	SeverityFatal
+	SeverityFatal2
+	SeverityFatal3
+	SeverityFatal4
+)
+```
+
+`Record` has `Attributes` and `AddAttributes` methods,
+like [`slog.Record.Attrs`](https://pkg.go.dev/log/slog#Record.Attrs)
+and [`slog.Record.AddAttrs`](https://pkg.go.dev/log/slog#Record.AddAttrs),
+in order to achieve high-performance when accessing and setting attributes efficiently.
+
+The `NewRecord(...RecordOption) (Record, error)` is a factory function
+used to create records using provided options.
+
+`Record` has a `Clone` method to allow copying records
+so that the SDK can offer concurrency safety.
+
+## Compatibility
+
+The backwards compatibility is achieved using the `embedded` design pattern
+that is already used in Trace API and Metrics API.
+
+## Benchmarking
+
+The benchmarks takes inspiration from [`slog`](https://pkg.go.dev/log/slog),
```

Comment:
We should add as an alternative (since people are bound to ask): Why can't we re-use slog.Record?

Reply:
I did my best here 892ee8f9

We can improve it if needed.